src/components/StakeSyrupModal/StakeSyrupModal.tsx

```diff
@@ -57,29 +57,31 @@ const StakeSyrupModal: React.FC<StakeSyrupModalProps> = ({
   const finalizedTransaction = useTransactionFinalizer();
   const userLiquidityUnstaked = useTokenBalance(
     account ?? undefined,
-    syrup.stakedAmount.token,
+    syrup.stakedAmount?.token,
   );
   const [typedValue, setTypedValue] = useState('');
   const [stakePercent, setStakePercent] = useState(0);
   const [approving, setApproving] = useState(false);
   const maxAmountInput = maxAmountSpend(userLiquidityUnstaked);
   const { parsedAmount, error } = useDerivedSyrupInfo(
     typedValue,
-    syrup.stakedAmount.token,
+    syrup.stakedAmount?.token,
     userLiquidityUnstaked,
   );
 
   const parsedAmountWrapped = wrappedCurrencyAmount(parsedAmount, chainId);
 
-  let hypotheticalRewardRate: TokenAmount = new TokenAmount(
-    syrup.rewardRate.token,
-    '0',
-  );
+  let hypotheticalRewardRate = syrup.rewardRate
+    ? new TokenAmount(syrup.rewardRate.token, '0')
```

Comment:
Why are we passing 0 as a string?

Reply:
This is for showing reward token as 0 when the value of staking input is 0.